### PR TITLE
Extend findAll with specification, pagable and entity graph to discard count query; Implement Slice<T> findSlice with hasNext check

### PR DESCRIPTION
**Pull request by StanislavValov**

### CODE REVIEW

**Comment by reda-alaoui** [comment]
I don't understand what is intended here. There is no explanation of what is done and why.
Why is the version changed? Where do those new methods come from? Are they exposed by `spring-data-jpa` and missing `EntityGraph` argument ?

**Comment by StanislavValov** [comment]
> I don't understand what is intended here. There is no explanation of what is done and why. Why is the version changed? Where do those new methods come from? Are they exposed by `spring-data-jpa` and missing `EntityGraph` argument ?

Hi, I extend findAll with option to skip count query executions to improve performance.

File: core/src/main/java/com/cosium/spring/data/jpa/entity/graph/repository/EntityGraphJpaSpecificationExecutor.java [hunk 1]
```diff
@@ -30,4 +31,6 @@
 
   /** @see JpaSpecificationExecutor#findAll(Specification, Sort) */
   List<T> findAll(Specification<T> spec, Sort sort, EntityGraph entityGraph);
+
+  Slice<T> findSlice(Specification<T> spec, Pageable pageable, EntityGraph entityGraph);
```
Comment (reda-alaoui) [inline core/src/main/java/com/cosium/spring/data/jpa/entity/graph/repository/EntityGraphJpaSpecificationExecutor.java:35]: Does this method minus the entity graph exists on spring-data-jpa?
If not, there is no reason to add it to this extension. People are free to extend the library with their own methods without needing to patch this extension.

Same remarks for the other method.

**Review by reda-alaoui — CHANGES_REQUESTED** [review]
The added methods seem to be out of this extension’s scope.

File: core/src/main/java/com/cosium/spring/data/jpa/entity/graph/repository/EntityGraphJpaSpecificationExecutor.java (hunk 1 above)
Comment (StanislavValov) [reply core/src/main/java/com/cosium/spring/data/jpa/entity/graph/repository/EntityGraphJpaSpecificationExecutor.java:35]: ok, I will close this PR then.

**Pull request closed without merging**